subsys/usb/usb_c/usbc_tc_src_states.c

```diff
@@ -0,0 +1,333 @@
+		usbc_timer_expired(&tc->tc_t_cc_debounce) == false) {
+		return;
+	}
+
+	/*
+	 * The port shall transition to Attached.SRC when VBUS is at vSafe0V
+	 * and the SRC.Rd state is detected on exactly one of the CC1 or CC2
+	 * pins for at least tCCDebounce.
+	 *
+	 * If the port supports Debug Accessory Mode, it shall transition to
+	 * UnorientedDebugAccessory.SRC when VBUS is at vSafe0V and the SRC.Rd
+	 * state is detected on both the CC1 and CC2 pins for at least
+	 * tCCDebounce.
+	 */
+	if (usbc_vbus_check_level(vbus, TC_VBUS_SAFE0V)) {
+		if (new_cc_state == TC_CC_UFP_ATTACHED) {
+			tc_set_state(dev, TC_ATTACHED_SRC_STATE);
+		}
+	}
+}
+
+void tc_attach_wait_src_exit(void *obj)
+{
+	struct tc_sm_t *tc = (struct tc_sm_t *)obj;
+
+	/* Stop debounce timer */
+	usbc_timer_stop(&tc->tc_t_cc_debounce);
+}
+
+/**
+ * @brief 4.5.2.2.9 Attached.SRC State
+ *
+ * When in the Attached.SRC state, the port is attached and operating as a
+ * Source. When the port initially enters this state it is also operating
+ * as a DFP. Subsequently, the initial power and data roles can be changed
+ * using USB PD commands.
+ *
+ * Requirements:
+ *  1: If the port needs to determine the orientation of the connector, it
+ *     shall do so only upon entry to the Attached.SRC state by detecting
+ *     which of the CC1 or CC2 pins is connected through the
+ *     cable, i.e., which CC pin is in the SRC.Rd state.
+ *     NOTE: Implemented in tc_attached_src_entry.
+ *
+ * 2: If the port has entered this state from the AttachWait.SRC state,
+ *    the SRC.Rd state will be on only one of the CC1 or CC2 pins. The
+ *    port shall source current on this CC pin and monitor its state.
+ *    NOTE: Implemented in the super state of AttachWait.SRC.
+ *
+ * 3: The port shall provide an Rp
+ *    NOTE: Implemented in the super state of AttachWait.SRC.
+ *
+ * 5: The port shall supply VBUS current at the level it advertises on Rp.
+ *    NOTE: Implemented in tc_attached_src_entry.
+ *
+ * 7: The port shall not initiate any USB PD communications until VBUS
+ *    reaches vSafe5V.
+ *    NOTE: Implemented in tc_attached_src_run.
+ *
+ * 8: The port may negotiate a USB PD PR_Swap, DR_Swap or VCONN_Swap.
+ *    NOTE: Implemented in tc_attached_src_run.
+ *
+ * 9: If the port supplies VCONN, it shall do so within t_VCONN_ON.
+ *    NOTE: Implemented in tc_attached_src_entry.
+ */
+
+void tc_attached_src_entry(void *obj)
+{
+	struct tc_sm_t *tc = (struct tc_sm_t *)obj;
+	const struct device *dev = tc->dev;
+	struct usbc_port_data *data = dev->data;
+	const struct device *tcpc = data->tcpc;
+
+	LOG_INF("Attached.SRC");
+
+	/* Initial data role for source is DFP */
+	tcpc_set_roles(tcpc, TC_ROLE_SOURCE, TC_ROLE_DFP);
+
+	/* Set cc polarity */
+	tcpc_set_cc_polarity(tcpc, tc->cc_polarity);
+
+	/* Start sourcing VBUS */
+	if (data->policy_cb_src_en(dev, true) == 0) {
+		/* Start sourcing VCONN */
+		if (policy_check(dev, CHECK_VCONN_CONTROL)) {
```

Comment:
Should VCONN support be its own config, or do you want to leave it entirely to policy?

Reply:
Let's leave it as policy for now. We can add it to the config in a future PR if we need to.